### PR TITLE
[WIP] Update to sdk 5 0

### PR DESCRIPTION
**Pull request by cgroschupp**
These changes require the PR profitbricks/profitbricks-sdk-go#35, please do not merge this PR before the other.

### CODE REVIEW

**Comment by cgroschupp** [comment]
@jasmingacic done

**Comment by jasmingacic** [comment]
@cgroschupp Meanwhile can you provide us with full `make testacc` output before while we are waiting on PR #34 of the SDK to be merged

**Comment by jasmingacic** [comment]
SDK pull requests have been merged and version has been bumped to v5.0.1.

**Comment by jasmingacic** [comment]
@cgroschupp I tried running `make testacc` on your PR but after several tries I keep getting this error:

```
$ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-profitbricks  [no test files]
=== RUN   TestAccDataSourceDatacenter_matching
--- PASS: TestAccDataSourceDatacenter_matching (38.58s)
=== RUN   TestAccDataSourceImage_basic
--- PASS: TestAccDataSourceImage_basic (11.01s)
=== RUN   TestAccDataSourceLocation_basic
--- PASS: TestAccDataSourceLocation_basic (5.01s)
=== RUN   TestAccResource_basic
--- FAIL: TestAccResource_basic (17.36s)
        testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

                * data.profitbricks_resource.res: data.profitbricks_resource.res: There are no resources that match the search criteria
        testing.go:494: Error destroying resource! WARNING: Dangling resources
                may exist. The full state and error is shown below.

                Error: Error refreshing: 1 error(s) occurred:

                * data.profitbricks_resource.res: 1 error(s) occurred:

                * data.profitbricks_resource.res: data.profitbricks_resource.res: There are no resources that match the search criteria

                State: <nil>
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccProfitBricksDataCenter_Basic
--- FAIL: TestAccProfitBricksDataCenter_Basic (59.47s)
        testing.go:494: Error destroying resource! WARNING: Dangling resources
                may exist. The full state and error is shown below.

                Error: Check failed: DataCenter still exists 576557f3-313a-42c0-a008-e8cf3f4b01cf HTTP Status: 404
                Error Messages:Error Code: 309 Message: Resource does not exist


                State: <no state>
=== RUN   TestAccProfitBricksFirewall_Basic
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x153441a]

goroutine 2027 [running]:
github.com/terraform-providers/terraform-provider-profitbricks/profitbricks.resourceProfitBricksNicRead(0xc4202e0d20, 0x1689fe0, 0xc42046c100, 0x0, 0x0)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/profitbricks/resource_profitbricks_nic.go:119 +0x34a
github.com/terraform-providers/terraform-provider-profitbricks/profitbricks.resourceProfitBricksNicCreate(0xc4202e0d20, 0x1689fe0, 0xc42046c100, 0x0, 0xc4205eace8)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/profitbricks/resource_profitbricks_nic.go:104 +0x4a0
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc420066b40, 0xc4203ae370, 0xc4200eb700, 0x1689fe0, 0xc42046c100, 0xc420384101, 0x3d, 0x0)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:193 +0x3b6
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc4201d4310, 0xc42031c0a0, 0xc4203ae370, 0xc4200eb700, 0x1, 0x54, 0x13)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:259 +0xa4
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.(*EvalApply).Eval(0xc4203788c0, 0x19daac0, 0xc420322380, 0x2, 0x2, 0x168c3e1, 0x4)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/eval_apply.go:57+0x239
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.EvalRaw(0x19cf120, 0xc4203788c0, 0x19daac0, 0xc420322380, 0x0, 0x0, 0x0, 0x0)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/eval.go:53 +0x173
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc4200eace0, 0x19daac0,0xc420322380, 0x2, 0x2, 0x168c3e1, 0x4)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/eval_sequence.go:14 +0x7e
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.EvalRaw(0x19cfae0, 0xc4200eace0, 0x19daac0, 0xc420322380, 0x15cd400, 0x19db8a2, 0x15b5e20, 0xc420218440)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/eval.go:53 +0x173
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.Eval(0x19cfae0, 0xc4200eace0, 0x19daac0, 0xc420322380, 0xc4200eace0, 0x19cfae0, 0xc4200eace0, 0x107b234)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/eval.go:34 +0x4d
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x1668be0, 0xc4201da060, 0x0, 0x0)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/terraform/graph.go:126 +0xd0d
github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc420188a80, 0x1668be0, 0xc4201da060, 0xc42034ab00)
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/dag/walk.go:387 +0x3c1
created by github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/dag.(*Walker).Update
        /Volumes/Data/go/src/github.com/terraform-providers/terraform-provider-profitbricks/vendor/github.com/hashicorp/terraform/dag/walk.go:310 +0x1364
FAIL    github.com/terraform-providers/terraform-provider-profitbricks/profitbricks     288.154s
make: *** [testacc] Error 1
```

The CI ran tests on latest version of the code few hours ago and it worked fine.

**Comment by cgroschupp** [comment]
@jasmingacic I'm working on it.

**Comment by denza** [comment]
@cgroschupp Could you please merge PR https://github.com/cgroschupp/terraform-provider-profitbricks/pull/1

**Comment by cgroschupp** [comment]
@denza this fix is not only relevant for my branch, please create a PR against the main repo. I need some time to fix the tests.

File: profitbricks/resource_profitbricks_volume.go [hunk 1]
```diff
@@ -157,57 +159,60 @@ func resourceProfitBricksVolumeCreate(d *schema.ResourceData, meta interface{})
...
-	_, errState := getStateChangeConf(meta, d, volume.Headers.Get("Location"), schema.TimeoutCreate).WaitForState()
+	_, errState := getStateChangeConf(meta, d, resp.Headers.Get("Location"), schema.TimeoutCreate).WaitForState()
 	if errState != nil {
 		return errState
 	}
 
-	volume = profitbricks.AttachVolume(dcId, serverId, volume.Id)
-	if volume.StatusCode > 299 {
-		return fmt.Errorf("An error occured while attaching a volume dcId: %s server_id: %s ID: %s Response: %s", dcId, serverId, volume.Id, volume.Response)
+	resp, err = connection.AttachVolume(dcId, serverId, resp.ID)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_volume.go]: `resp` -> `volume`

File: profitbricks/resource_profitbricks_volume.go [hunk 2]
```diff
@@ -157,57 +159,60 @@ func resourceProfitBricksVolumeCreate(d *schema.ResourceData, meta interface{})
...
+		volume.Properties.SSHKeys = nil
 	}
 
 	if _, ok := d.GetOk("availability_zone"); ok {
 		raw := d.Get("availability_zone").(string)
 		volume.Properties.AvailabilityZone = raw
 	}
 
-	volume = profitbricks.CreateVolume(dcId, volume)
+	resp, err := connection.CreateVolume(dcId, volume)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_volume.go]: `resp` -> `volume`

File: profitbricks/resource_profitbricks_snapshot.go [hunk 3]
```diff
@@ -94,26 +94,34 @@ func resourceProfitBricksSnapshotUpdate(d *schema.ResourceData, meta interface{}
...
+		status, err = connection.GetSnapshot(d.Id())
+
+		if err != nil {
+			return fmt.Errorf("An error occured while fetching a snapshot ID %s %s", d.Id(), err)
+		}
 	}
 
 	dcId := d.Get("datacenter_id").(string)
-	dc := profitbricks.GetDatacenter(dcId)
+	dc, _ := connection.GetDatacenter(dcId)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_snapshot.go]: Error checking would be good here

File: profitbricks/resource_profitbricks_user.go [hunk 4]
```diff
@@ -151,18 +156,23 @@ func resourceProfitBricksUserUpdate(d *schema.ResourceData, meta interface{}) er
...
+	resp, err := connection.DeleteUser(d.Id())
+	if err != nil {
 		//try again in 20 seconds
 		time.Sleep(20 * time.Second)
-		resp = profitbricks.DeleteUser(d.Id())
-		if resp.StatusCode > 299 && resp.StatusCode != 404 {
-			return fmt.Errorf("An error occured while deleting a user %s %s", d.Id(), string(resp.Body))
+		resp, err = connection.DeleteUser(d.Id())
+		if err != nil {
+			if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_user.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_volume.go [hunk 5]
```diff
@@ -157,57 +159,60 @@ func resourceProfitBricksVolumeCreate(d *schema.ResourceData, meta interface{})
...
 
-	volume := profitbricks.GetVolume(dcId, d.Id())
+	volume, err := connection.GetVolume(dcId, d.Id())
 
-	if volume.StatusCode > 299 {
-		if volume.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_volume.go]: `err2`-> `apiError`

File: profitbricks/resource_profitbricks_firewall.go [hunk 6]
```diff
@@ -129,39 +130,42 @@ func resourceProfitBricksFirewallCreate(d *schema.ResourceData, meta interface{}
...
-
-	if fw.StatusCode > 299 {
-		if fw.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	fw, err := connection.GetFirewallRule(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Get("nic_id").(string), d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_firewall.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_group.go [hunk 7]
```diff
@@ -131,23 +132,30 @@ func resourceProfitBricksGroupCreate(d *schema.ResourceData, meta interface{}) e
...
-
-	if group.StatusCode > 299 {
-		if group.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	group, err := connection.GetGroup(d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_group.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_ipblock.go [hunk 8]
```diff
@@ -42,45 +42,49 @@ func resourceProfitBricksIPBlock() *schema.Resource {
...
-
-	if ipblock.StatusCode > 299 {
-		if ipblock.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	ipblock, err := connection.GetIPBlock(d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipblock.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_ipfailover.go [hunk 9]
```diff
@@ -65,46 +66,50 @@ func resourceProfitBricksLanIPFailoverCreate(d *schema.ResourceData, meta interf
...
-
-	if lan.StatusCode > 299 {
-		if lan.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	lan, err := connection.GetLan(d.Get("datacenter_id").(string), d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipfailover.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_ipfailover.go [hunk 10]
```diff
@@ -117,21 +122,27 @@ func resourceProfitBricksLanIPFailoverUpdate(d *schema.ResourceData, meta interf
...
+	if err != nil {
 		//try again in 90 seconds
 		time.Sleep(90 * time.Second)
-		resp = profitbricks.PatchLan(dcid, lanid, *properties)
-		if resp.StatusCode > 299 && resp.StatusCode != 404 {
-			return fmt.Errorf("An error occured while removing a lans ipfailover groups dcId %s ID %s %s", d.Get("datacenter_id").(string), d.Id(), string(resp.Response))
+		resp, err = connection.UpdateLan(dcid, lanid, *properties)
+
+		if err != nil {
+			if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipfailover.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_ipfailover.go [hunk 11]
```diff
@@ -117,21 +122,27 @@ func resourceProfitBricksLanIPFailoverUpdate(d *schema.ResourceData, meta interf
...
 
 	//remove the failover group
 	properties := &profitbricks.LanProperties{
-		IpFailover: &[]profitbricks.IpFailover{},
+		IPFailover: &[]profitbricks.IPFailover{},
 	}
 
-	resp := profitbricks.PatchLan(dcid, lanid, *properties)
-	if resp.StatusCode > 299 {
+	resp, err := connection.UpdateLan(dcid, lanid, *properties)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipfailover.go]: `resp` -> `lan`

File: profitbricks/resource_profitbricks_lan.go [hunk 12]
```diff
@@ -62,29 +63,33 @@ func resourceProfitBricksLanCreate(d *schema.ResourceData, meta interface{}) err
...
-
-	if lan.StatusCode > 299 {
-		if lan.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	lan, err := connection.GetLan(d.Get("datacenter_id").(string), d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_lan.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_lan.go [hunk 13]
```diff
@@ -110,19 +115,25 @@ func resourceProfitBricksLanUpdate(d *schema.ResourceData, meta interface{}) err
...
+	if err != nil {
 		//try again in 120 seconds
 		time.Sleep(120 * time.Second)
-		resp = profitbricks.DeleteLan(dcId, d.Id())
-		if resp.StatusCode > 299 && resp.StatusCode != 404 {
-			return fmt.Errorf("An error occured while deleting a lan dcId %s ID %s %s", d.Get("datacenter_id").(string), d.Id(), string(resp.Body))
+		resp, err = connection.DeleteLan(dcId, d.Id())
+
+		if err != nil {
+			if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_lan.go]: `err2`-> `apiError`

File: profitbricks/resource_profitbricks_loadbalancer.go [hunk 14]
```diff
@@ -63,50 +63,54 @@ func resourceProfitBricksLoadbalancerCreate(d *schema.ResourceData, meta interfa
...
-
-	if lb.StatusCode > 299 {
-		if lb.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	lb, err := connection.GetLoadbalancer(d.Get("datacenter_id").(string), d.Id())
+
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_loadbalancer.go]: `err2` => `apiError`

File: profitbricks/resource_profitbricks_nic.go [hunk 15]
```diff
@@ -88,40 +89,46 @@ func resourceProfitBricksNicCreate(d *schema.ResourceData, meta interface{}) err
...
 func resourceProfitBricksNicRead(d *schema.ResourceData, meta interface{}) error {
-	nic := profitbricks.GetNic(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Id())
-	if nic.StatusCode > 299 {
-		if nic.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	connection := meta.(*profitbricks.Client)
+	nic, err := connection.GetNic(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Id())
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_nic.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_server.go [hunk 16]
```diff
@@ -511,26 +530,32 @@ func resourceProfitBricksServerCreate(d *schema.ResourceData, meta interface{})
...
 	serverId := d.Id()
 
-	server := profitbricks.GetServer(dcId, serverId)
-	if server.StatusCode > 299 {
-		if server.StatusCode == 404 {
-			d.SetId("")
-			return nil
+	server, err := connection.GetServer(dcId, serverId)
+	if err != nil {
+		if err2, ok := err.(profitbricks.ApiError); ok {
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_server.go]: `err2` -> `apiError`

File: profitbricks/resource_profitbricks_firewall.go [hunk 17]
```diff
@@ -215,10 +220,10 @@ func resourceProfitBricksFirewallUpdate(d *schema.ResourceData, meta interface{}
 		properties.IcmpCode = &tempIcmpCode
 	}
 
-	resp := profitbricks.PatchFirewallRule(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Get("nic_id").(string), d.Id(), properties)
+	resp, err := connection.UpdateFirewallRule(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Get("nic_id").(string), d.Id(), properties)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_firewall.go]: `resp` -> `fireWall` or `fw`

File: profitbricks/resource_profitbricks_ipfailover.go [hunk 18]
```diff
@@ -117,21 +122,27 @@ func resourceProfitBricksLanIPFailoverUpdate(d *schema.ResourceData, meta interf
...
+	if err != nil {
 		//try again in 90 seconds
 		time.Sleep(90 * time.Second)
-		resp = profitbricks.PatchLan(dcid, lanid, *properties)
-		if resp.StatusCode > 299 && resp.StatusCode != 404 {
-			return fmt.Errorf("An error occured while removing a lans ipfailover groups dcId %s ID %s %s", d.Get("datacenter_id").(string), d.Id(), string(resp.Response))
+		resp, err = connection.UpdateLan(dcid, lanid, *properties)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipfailover.go]: `resp` -> `lan`

File: profitbricks/resource_profitbricks_ipfailover_test.go [hunk 19]
```diff
@@ -83,22 +84,23 @@ func testAccCheckLanIPFailoverGroupExists(n string, lan *profitbricks.Lan, failo
...
 
 func testAccCheckDProfitBricksLanIPFailoverDestroyCheck(s *terraform.State) error {
+	connection := testAccProvider.Meta().(*profitbricks.Client)
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "profitbricks_ipfailover" {
 			continue
 		}
 		nicUuid := rs.Primary.Attributes["nicuuid"]
-		resp := profitbricks.GetLan(rs.Primary.Attributes["datacenter_id"], rs.Primary.Attributes["lan_id"])
+		resp, _ := connection.GetLan(rs.Primary.Attributes["datacenter_id"], rs.Primary.Attributes["lan_id"])
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_ipfailover_test.go]: `resp` -> `lan`

File: profitbricks/resource_profitbricks_loadbalancer.go [hunk 20]
```diff
@@ -63,50 +63,54 @@ func resourceProfitBricksLoadbalancerCreate(d *schema.ResourceData, meta interfa
 		},
 	}
 
-	lb = profitbricks.CreateLoadbalancer(d.Get("datacenter_id").(string), lb)
+	resp, err := connection.CreateLoadbalancer(d.Get("datacenter_id").(string), lb)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_loadbalancer.go]: `resp` -> `lb`

File: profitbricks/resource_profitbricks_share_test.go [hunk 21]
```diff
@@ -36,10 +36,11 @@ func TestAccProfitBricksShare_Basic(t *testing.T) {
 }
 
 func testAccCheckDProfitBricksShareDestroyCheck(s *terraform.State) error {
+	connection := testAccProvider.Meta().(*profitbricks.Client)
 	for _, rs := range s.RootModule().Resources {
-		resp := profitbricks.GetShare(rs.Primary.Attributes["group_id"], rs.Primary.Attributes["resource_id"])
+		resp, err := connection.GetShare(rs.Primary.Attributes["group_id"], rs.Primary.Attributes["resource_id"])
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_share_test.go]: `resp` -> `share`

File: profitbricks/resource_profitbricks_datacenter.go [hunk 22]
```diff
@@ -104,7 +110,11 @@ func resourceProfitBricksDatacenterUpdate(d *schema.ResourceData, meta interface
 		return fmt.Errorf("Data center is created in %s location. You can not change location of the data center to %s. It requires recreation of the data center.", oldLocation, newLocation)
 	}
 
-	resp := profitbricks.PatchDatacenter(d.Id(), obj)
+	resp, err := connection.UpdateDataCenter(d.Id(), obj)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_datacenter.go]: `resp` -> `dc`

File: profitbricks/resource_profitbricks_firewall.go [hunk 23]
```diff
@@ -129,39 +130,42 @@ func resourceProfitBricksFirewallCreate(d *schema.ResourceData, meta interface{}
 		fw.Properties.IcmpCode = &tempIcmpCodee
 	}
 
-	fw = profitbricks.CreateFirewallRule(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Get("nic_id").(string), fw)
+	resp, err := connection.CreateFirewallRule(d.Get("datacenter_id").(string), d.Get("server_id").(string), d.Get("nic_id").(string), fw)
```
Comment (jasmingacic) [inline profitbricks/resource_profitbricks_firewall.go]: `resp` -> `fw`

File: profitbricks/provider.go [hunk 24]
```diff
@@ -29,7 +29,7 @@ func Provider() terraform.ResourceProvider {
 			"endpoint": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("PROFITBRICKS_API_URL", profitbricks.Endpoint),
+				DefaultFunc: schema.EnvDefaultFunc("PROFITBRICKS_API_URL", ""),
```
[the remark this answers is not in the document]
Comment (cgroschupp) [reply profitbricks/provider.go]: ok, i rebased the branch.

**Comment by jasmingacic** [comment]
The best way to test individual acceptance tests is:
```
make testacc TEST=./profitbricks/  TESTARGS='-run=[test_name]'
```

```
TestAccProfitBricksUser_Basic
=== RUN   TestAccProfitBricksUser_Basic
--- PASS: TestAccProfitBricksUser_Basic (231.13s)
PASS
ok      terraform-provider-profitbricks/profitbricks    231.136s
```
```
go test -v --run TestAccProfitBricksGroup_Basic
=== RUN   TestAccProfitBricksGroup_Basic
--- PASS: TestAccProfitBricksGroup_Basic (55.25s)
PASS
ok      terraform-provider-profitbricks/profitbricks    55.253s
```

p.s. still waiting on shares to complete

**Comment by jasmingacic** [comment]
```
TestAccProfitBricksShare_Basic
=== RUN   TestAccProfitBricksShare_Basic
--- PASS: TestAccProfitBricksShare_Basic (98.89s)
PASS
ok      terraform-provider-profitbricks/profitbricks    98.897s
```

File: profitbricks/provider.go (hunk 24 above)
[the remark this answers is not in the document]
Comment (jasmingacic) [reply profitbricks/provider.go]: Regardless of rebasing just re-add default URL value
```
DefaultFunc: schema.EnvDefaultFunc("PROFITBRICKS_API_URL", profitbricks.Endpoint),
```

**Comment by jasmingacic** [comment]
@cgroschupp so far this is looking good. After you fixed the URL piece this is good to be merged.

Thanks

File: profitbricks/provider.go (hunk 24 above)
[the remark this answers is not in the document]
Comment (cgroschupp) [reply profitbricks/provider.go]: this is not more possible: https://github.com/profitbricks/profitbricks-sdk-go/blob/master/req.go#L39
profitbricks.Endpoint no longer exists.

**Comment by jasmingacic** [comment]
@cgroschupp Sorry you are right it is no longer supported. If no value is provided the SDK will use the latest version.

**Comment by jasmingacic** [comment]
Next version of the binary will be 1.3.3, and you can expect to see it [here](https://releases.hashicorp.com/terraform-provider-profitbricks/).

**Pull request merged**